File: w09/d03/quin_cogdell/mail_objects_app/spec/javascripts/package_spec.js
// Package should have a content array
// Package should have a fragile boolean
// Package should have a broken boolean
// Package should have a method shake()
// If the Package's fragile is true and shake() is invoked then broken should be set to true
// Package should have an isBroken() method that returns whether it is broken or not

describe("a Package", function(){

  beforeEach(function(){
    package = new Package();
    item1 = "present";
    item2 = "office supplies";
  });

  it("has a content array", function(){
    package.addContent(item1);
    package.addContent(item2);
    expect(package.content).toEqual([item1, item2]);
  });

  it("has a fragile boolean", function(bool){
    package.setFragile(true);
    expect(package.fragile).toBe(true);
  });

  it("has a broken property that is initialized to false", function() {
    expect(package.broken).toEqual(false);
  });

  it("has an isBroken() method that returns whether it is broken or not", function() {
    package.broken = true;
    expect(package.isBroken()).toEqual(true);
  });

  it("has a method shake that sets broken to true if it is fragile", function(){
    package.setFragile(true);
    package.shake();
    expect(package.isBroken()).toEqual(true);
  });

  it("has a isHeavy method", function(){
    spyOn(package, 'isHeavy').andCallThrough();
    var answer = package.isHeavy();
    expect(package.isHeavy).toHaveBeenCalled();
    expect(answer).toBe(true);
  });

});

File: w09/d01/edward_shin/mail_it/spec/package_spec.rb
require_relative 'spec_helper'
require_relative '../package'

describe Package do

  it "should have inherit mail_item" do
    package = Package.new
    expect(package.stamp_value).to be_true
  end

  it "should have a content array" do
    package = Package.new
    expect(package.content.class).to eq(Array)
  end

  it "has an is_fragile boolean" do
    package = Package.new
    expect(package.is_fragile).to eq(false)
  end

  it "should have a is_broken boolean" do
    package = Package.new
    expect(package.is_broken).to eq(false)
  end

  it "should have a method shake that when shaken, is_broken is set to true" do
    package = Package.new
    package.shake
    expect(package.is_broken).to eq(true)
  end

end










